torchtyping/typechecker.py

```diff
@@ -301,7 +300,7 @@ def check_type(*args, **kwargs):
             # Now check if it's annotating a tensor
             if is_torchtyping_annotation:
                 base_cls, *all_metadata = get_args(expected_type)
-                if not issubclass(base_cls, torch.Tensor):
+                if not isinstance(base_cls(), TensorLike):
```

Comment:
I'm not sure about this change. As mentioned, the protocol class only supports isintance() because it has properties. This means I had to require default construction.
But, I think this test may be unnecessary - after all the other tests I think we know this is a TensorLike element? 
I think it might be better to just get rid of this test.

Reply:
In fact, it does seem I have a strong motivation to remove this. The case where I want to apply it is to check shape signatures on an abstract base class so default construction may not be an option.

Reply:
I have updated the PR accordingly.